src/index.ts

```diff
@@ -0,0 +1,189 @@
+import type appdmgType from 'appdmg';
+import debugCreator from 'debug';
+import { existsSync, promises as fs } from 'fs';
+import * as path from 'path';
+
+const debug = debugCreator('electron-installer-dmg');
+
+export type ElectronInstallerDMGOptions = {
+  /**
+   * The `.app` directory generated by Electron Packager.
+   */
+  appPath: string;
+  /**
+   * The application name.
+   */
+  name: string;
+  /**
+   * The title of the produced DMG, which will be shown when mounted.
+   *
+   * Defaults to the application name.
+   */
+  title?: string;
+  /**
+   * Path to the background for the DMG window. Background image should be of size 658x498.
+   */
+  background?: string;
+  /**
+   * Path to the icon to use for the app in the DMG window.
+   */
+  icon?: string;
+  /**
+   * How big to make the icon for the app in the DMG. [Default: `80`].
+   */
+   iconSize?: number;
+  /**
+   * Overwrite an existing DMG file if if already exists.
+   */
+  overwrite?: boolean;
+  /**
+   * The content that will appear in the window when user opens the `.dmg` file.
+   *
+   * Defaults to an array of two icons, the application and the /Applications destination folder.
+   */
+  contents?: appdmgType.SpecificationContents[] | ((opts: ElectronInstallerDMGOptions) => appdmgType.SpecificationContents[]);
+  /**
+   * Disk image format. [Default: `UDZO`]
+   *
+   * Must be one of the following:
+   *  - `UDRW` :arrow_right: read/write image
+   *  - `UDRO` :arrow_right: read-only image
+   *  - `UDCO` :arrow_right: ADC-compressed image
+   *  - `UDZO` :arrow_right: zlib-compressed image
+   *  - `UDBZ` :arrow_right: bzip2-compressed image
+   *  - `ULFO` :arrow_right: lzfse-compressed image (macOS 10.11+ only)
+   */
+  format?: 'UDRW' | 'UDRO' | 'UDCO' | 'UDZO' | 'UDBZ' | 'ULFO';
+  /**
+   * Additional options to pass through to [`appdmg`](https://npm.im/appdmg)
+   *
+   * You can use this to set additional features like `background-color` and
+   * `code-sign`.  See the docs of the `appdmg` module for all possible options.
+   */
+  additionalDMGOptions?: Omit<appdmgType.Specification, 'title' | 'contents' | 'icon' | 'icon-size' | 'background' | 'format'>;
+} & ({
+  /**
+   * The directory to put the DMG into. [Default: `process.cwd()`].
+   */
+  out?: string;
+} | {
+  /**
+   * The full path to write the DMG to.
+   */
+  dmgPath: string;
```

Comment:
This should be an optional parameter I think
```suggestion
  dmgPath?: string;
```